src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -353,11 +353,16 @@ template <class ImplClass>
 CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_AttachToThreadNetwork(
     const Thread::OperationalDataset & dataset, NetworkCommissioning::Internal::WirelessDriver::ConnectCallback * callback)
 {
-    // Reset the previously set callback since it will never be called in case incorrect dataset was supplied.
-    mpConnectCallback = nullptr;
-    ReturnErrorOnFailure(Impl()->SetThreadEnabled(false));
-    ReturnErrorOnFailure(Impl()->SetThreadProvision(dataset.AsByteSpan()));
-
+    Thread::OperationalDataset current_dataset;
+    // Validation the dataset change with the current state
+    ThreadStackMgrImpl().GetThreadProvision(current_dataset);
+    if (!dataset.AsByteSpan().data_equal(current_dataset.AsByteSpan()))
```

Comment:
Hmm, maybe you could just do sth like:
```
if (dataset.AsByteSpan().data_equal(current_dataset.AsByteSpan() && callback == nullptr)
{
   return CHIP_NO_ERROR;
}
```
1. If the callback is provided, we probably want to restart the interface. Otherwise, the callback will never be called, so if a user called `ConnectNetwork` for the second time, it would fail (I don't know why a user would want to do it, but it should probably work).
2. If the callback is not provided we don't need to execute `if (dataset.IsCommissioned())` code.